### PR TITLE
fix flaky tests + go.mod cleanup

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

* fix flaky tests
* cleanup go.mod dependency
* fix typo in invoke scripts

### Motivation

Severals flaky tests reported by the CI

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Review by vboulineau — DISMISSED** [review]

File: pkg/util/kubernetes/apiserver/metadata_controller_test.go [hunk 1]
```diff
@@ -430,15 +432,25 @@ func TestMetadataController(t *testing.T) {
 	informerFactory.Start(stop)
 	go metaController.Run(stop)
 
-	for !metaController.endpointsListerSynced() && !metaController.nodeListerSynced() {
-		time.Sleep(150 * time.Millisecond)
-	}
+	retry.DoWithRetry(t, "wait Lister Synced", 10, 50*time.Millisecond, func() (string, error) {
```
Comment (olivielpeau) [inline pkg/util/kubernetes/apiserver/metadata_controller_test.go:435]: interesting, this is very similar to our own testutil here: https://github.com/DataDog/datadog-agent/blob/7f5ebddd1a3d36e705c59262ee02faf2ecf1a2b4/pkg/util/testutil/timeout.go#L29

It'd be good to use one or the other across the test codebase, but I guess we can address this later. `gruntwork-io/terratest` looks interesting, there are other test utils there that could prove useful for other tests.

**Review by olivielpeau — COMMENTED** [review]
* are the dep upgrades in `go.mod` intended? If not I'd rather not include them for 7.20 (since we're after the code freeze)
* are there any actual semantic changes in the `LICENSE-3rdparty.csv` file? Asking mostly to know if there are cases where the linter that's supposed to check this file doesn't work 100% :)

File: pkg/util/kubernetes/apiserver/metadata_controller_test.go (hunk 1 above)
Comment (clamoriniere) [reply pkg/util/kubernetes/apiserver/metadata_controller_test.go:435]: I didn't know the `assertTrueBeforeTimeout` I will use it.

File: go.mod [hunk 2]
```diff
@@ -172,7 +172,7 @@ replace (
 	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191016114939-2b2b218dc1df
 	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191016114407-2e83b6f20229
 	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191016114748-65049c67a58b
-	k8s.io/kube-state-metrics => github.com/clamoriniere/kube-state-metrics v1.8.1-0.20200412142917-6c764c23fffb
+	k8s.io/kube-state-metrics => k8s.io/kube-state-metrics v1.9.6-0.20200413182837-dbbe062e36a4
```
Comment (clamoriniere) [inline go.mod:175]: others dependencies update are coming from this one

File: pkg/util/kubernetes/apiserver/metadata_controller_test.go (hunk 1 above)
Comment (clamoriniere) [reply pkg/util/kubernetes/apiserver/metadata_controller_test.go]: done

**Review by olivielpeau — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (CharlyF) [reply go.mod:175]: 😬  sorry

File: pkg/util/kubernetes/apiserver/wpa_controller_test.go [hunk 3]
```diff
@@ -151,7 +157,6 @@ func TestUpdateWPA(t *testing.T) {
 }
```
Comment (CharlyF) [inline pkg/util/kubernetes/apiserver/wpa_controller_test.go:158]:
```suggestion
# newFakeWPAController creates an AutoscalersController. Use enableWPA(wpa_informers.SharedInformerFactory) to add the event handlers to it. Use Run() to add the event handlers and start processing the events.
```

**Review by CharlyF — APPROVED** [review]
LGTM, just one suggestion so it's clear when to use enableWPA or Run in the future.

**Review by albertvaka — APPROVED** [review]

**Pull request merged**